tests/Driver/OCI8/DriverTest.php

```diff
@@ -20,6 +20,13 @@ public function testPersistentAndExclusiveAreMutuallyExclusive(): void
         ]);
     }
 
+    public function testParseValidProtocol(): void
+    {
+        (new Driver())->connect([
+            'driverOptions' => ['protocol' => 'TCPS'],
+        ]);
```

Comment:
It would be better to have a functional test here. Is it possible to query some system view and get the protocol of the current connection? Otherwise, this test doesn't prove anything.

Reply:
@morozov I can check, but if you got the protocol wrong, it would not connect in the first place, so would I not be creating a test that would in theory never succeed?

Reply:
I'm talking about testing the happy path. If the user provides a valid value, the connection should be established using this protocol.

Reply:
@morozov OK. looking into it.

Reply:
@morozov The closest I can find in Oracle is via v$session_conn_info:

select * from V$SESSION_CONNECT_INFO;

` (DESCRIPTION=(address=(protocol=tcps)(port=1521)(host=1.1.1.1))(CONNECT_DATA=(CID=(PROGRAM=DataGrip)(HOST=__jdbc__)(USER=xxxx))(service_name=xxxxx.oraclecloud.com)(CONNECTION_ID=xxxxxxx)(SERVER=dedicated)(INSTANCE_NAME=xxxx))(ADDRESS=(PROTOCOL=TCP)(HOST=1.1.1.1)(PORT=1521)(publish_service_acl=true))(SOURCE_ROUTE=YES)(HOP_COUNT=2))
`
I would have to parse that string then to get the answer unless a pattern match would be OK. I'm working on the 'existence' of protocol=tcps - because there may be some magic in Oracle cloud where it is likely passing via TCP internally.

What are your thoughts?

Reply:
I think it's okay. If Oracle reports this parameter back, it means that it's passed to Oracle and it's valid.

> I'm working on the 'existence' of protocol=tcps

This is fine. I would structure the test this way:
1. Establish a connection w/ the default protocol. Check that the connection sting doesn't contain this substring.
2. Establish a connection w/ the other protocol. Check that the connection sting contains this substring.

This way, we will make sure that the substring appears in the connection string as a result of configuration change, not on its own. Also, those will be two independent tests (or one with a data provider).

Reply:
@morozov i will look shortly

Reply:
@morozov I've given it a shot, there is an issue though

* If it works via TCPS, chances are it won't connect using TCP
* Earlier versions (sub version 21) don't contain the connect string column in the view. 

This works in my 'successful' test case, on the right database version

`   $parameters            = ['envContext' => 'userEnv', 'sidContext' => 'sid', 'protocolString' => '%protocol=tcps%', 'networkServiceBanner' => '%tcp/ip%'];
        $tcpProtocolCheckQuery = 'with dbdetails as (select sys_context(:envContext, :sidContext) as sid from dual) SELECT 1 FROM v$session_connect_info sci join dbdetails on dbdetails.sid = sci.SID where lower(sci.client_connection_string) like :protocolString and lower(sci.network_service_banner) like :networkServiceBanner';
`

Reply:
Got it. In this case, instead of adding a test here, please update this test: https://github.com/doctrine/dbal/blob/5a9b576741cba40dfd9f63af0ead729ad825bdeb/tests/Driver/AbstractOracleDriver/EasyConnectStringTest.php#L49-L59

Add the new parameter to the input and update the expected value accordingly.

Reply:
Hey @morozov  (Thanks for your help by the way) - I have updated the tests accordingly and tested the test.

Reply:
@morozov oh - and I created a separate test rather than extend that test, is that ok?

Reply:
@morozov Just checking if there is anything else you want me to do here  - or we're all good?

Reply:
Yeah, this is fine too. Please squash all commits, and we can merge this.

Reply:
@morozov First time 'squashing' - I do hope I got it right.

Reply:
I don't think so, but here is a guide:



1. `git rebase -i origin/4.2.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already pushed. Don't forget the `--force` option otherwise git will try to merge both things together.

Reply:
@greg0ire I believe I have it this time - although the message is a bit skewiff with the note about 'combination of 2 commits'. Says 1 commit now :)

Reply:
Regarding the number of commits, we're good now, congrats!

Regarding the message itself, I don't know how you managed to do that :rofl: 

Can tell me what `git config --get 'commit.cleanup'` prints? I suspect an unexpected configuration on your end.

If I run `git commit --amend`, the comments disappear, as expected.
Can you try `git commit --amend --no-edit --cleanup=strip && git push --force`?

Reply:
@greg0ire, thanks for guiding @martinedge here. If it takes too much effort on your end, and if the author doesn't mind, we could amend the commit ourselves.

Reply:
@morozov I don't mind, if you want me to work through the process, I'm happy to either way, as it might save me mucking it up in the future :-) Up to you all, thankyou for your support!

Reply:
I did the regular `git commit --amend && git push --force`.